### PR TITLE
Modifying DTA  and DTL tasks for porting to Update 3

### PR DESCRIPTION
**Pull request by prawalagarwal**
[https://mseng.visualstudio.com/DefaultCollection/VSOnline/Test%20Infrastructure/_git/VSO/pullrequest/99378?view=commits](url)

### CODE REVIEW

File: Tasks/DeployAzureResourceGroup/task.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "version": {
         "Major": 1,
         "Minor": 0,
-        "Patch": 55
+        "Patch": 56
```
Comment (kmkumaran) [inline Tasks/DeployAzureResourceGroup/task.json]: should there be a minimum agent version check to ensure the u3 agent that supports updated Register-Environment is picked?

File: Tasks/DeployAzureResourceGroup/Utility.ps1 [hunk 2]
```diff
@@ -483,7 +483,7 @@ function Instantiate-Environment
     $machineSpecification = $resources -join ","
 
     Write-Verbose "Starting Register-Environment cmdlet call for resource group : $resourceGroupName" -Verbose
-    $environment = Register-Environment -EnvironmentName $outputVariable -EnvironmentSpecification $machineSpecification -WinRmProtocol "HTTPS" -Connection $connection -TaskContext $distributedTaskContext
+    $environment = Register-Environment -EnvironmentName $outputVariable -EnvironmentSpecification $machineSpecification -WinRmProtocol "HTTPS" -Connection $connection -TaskContext $distributedTaskContext -Persist
```
Comment (kmkumaran) [inline Tasks/DeployAzureResourceGroup/Utility.ps1]: DTA changes have the defensive check for the agent compat verifying if the parameter key is present. We added similar checks for VSTS. Are they required?

Comment (KrishnaAdityaB) [inline Tasks/DeployAzureResourceGroup/Utility.ps1]: We will add it if necessary to be defensive - after closing on the agent version

File: Tasks/DeployAzureResourceGroup/task.json (hunk 1 above)
Comment (KrishnaAdityaB) [inline Tasks/DeployAzureResourceGroup/task.json]: We have to close it. I am tracking this as a separate task on our storyboard.

**Pull request merged**